src/compas_model/elements/element.py

```diff
@@ -6,97 +6,134 @@
+    frame : :class:`compas.geometry.Frame`
+        The local coordinate frame of the element.
+    geometry : :class:`compas.datastructures.Mesh` | :class:`compas.geometry.Brep`, readonly
+        The geometry of the element, computed from the base shape and its features.
+    aabb : :class:`compas.geometry.Box`, readonly
         The Axis Aligned Bounding Box (AABB) of the element.
-    obb : :class:`compas.geometry.Box`
+    obb : :class:`compas.geometry.Box`, readonly
         The Oriented Bounding Box (OBB) of the element.
-    collision_mesh : :class:`compas.datastructures.Mesh`
+    collision_mesh : :class:`compas.datastructures.Mesh`, readonly
         The collision geometry of the element.
 
     """
 
     @property
     def __data__(self):
         # type: () -> dict
-        return {"geometry": self.geometry, "frame": self.frame, "name": self.name}
+        return {"frame": self.frame, "name": self.name}
+
+    def __init__(self, geometry=None, frame=None, name=None):
+        # type: (compas.geometry.Shape | compas.geometry.Brep | compas.datastructures.Mesh | None, compas.geometry.Frame | None, str | None) -> None
 
-    def __init__(self, geometry, frame=None, name=None):
-        # type: (compas.geometry.Geometry | compas.datastructures.Mesh, Frame | None, str | None) -> None
         super(Element, self).__init__(name=name)
         self.graph_node = None  # type: int | None
         self.tree_node = None  # type: ElementNode | None
-        self.geometry = geometry
-        self._frame = None
-        self._transformation = None
-        self._dimensions = []
         self._aabb = None
         self._obb = None
         self._collision_mesh = None
-        self.frame = frame
+        self._geometry = geometry
+        self._frame = frame
 
-    def __str__(self):
-        return "<Element with geometry {}>".format(self.geometry.__class__.__name__)
+    def __repr__(self):
+        # type: () -> str
+        return "Element(frame={!r}, name={})".format(self.frame, self.name)
 
-    # check the scene for a proper implementation of
-    # - frame
-    # - transformation
-    # - worldtransformation
-    # don't use them for now
-    # and assume world coordinates in all cases
+    def __str__(self):
+        # type: () -> str
+        return "<Element {}>".format(self.name)
 
     @property
     def frame(self):
+        # type: () -> compas.geometry.Frame | None
         return self._frame
 
     @frame.setter
     def frame(self, frame):
         self._frame = frame
 
-    @property
-    def transformation(self):
-        return self._transformation
+    # ==========================================================================
+    # Computed attributes
+    # ==========================================================================
 
-    @transformation.setter
-    def transformation(self, transformation):
-        self._transformation = transformation
+    # Who should store the transformations?
```

Comment:
I think there are several scenarios. For example:
a) When you use a model, you want to write `model.transform(...)` to transform the model.
b) If you work with groups of elements, you may want to visualize the transformation of these elements `model["my_group_something"].transform(...)`.
c) When no model data structure is used, and you work with a flat list of elements `element.transform(...)` including features as well (if they are geometries).